Automattic/jetpack
# RNA: Connection component

#### Changes proposed in this Pull Request:
- RNA directory and testing environment.
- The Jetpack's `AuthIframe` component is partially extracted into an RNA component.
- Jetpack build is temporarily adjusted to alias the `InPlaceConnection` component as `@automattic/jetpack-in-place-connection`.

#### Jetpack product discussion
p1HpG7-b01-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Build Jetpack using `yarn build`, confirm that it builds smoothly.
2. Connect Jetpack as a secondary user via In-Place Connection. Confirm that the user gets connected, and there are no errors in console.

#### Proposed changelog entry for your changes:
n/a
